Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -699,181 +699,209 @@ void doVayDepositionShapeN (const GetParticlePosition& GetPosition,
-                    - wq * invvol * invdt * 0.5_rt * sx_old[i] * sz_old[k]);
+                    - wq * invvol * invdt * 0.5_rt * sxo_szo);
 
                 // Jy
                 amrex::Gpu::Atomic::Add(&jy_arr(lo.x + i_new + i, lo.y + k_new + k, 0, 0),
-                    wqy * 0.25_rt * sx_new[i] * sz_new[k]);
+                    wqy * 0.25_rt * sxn_szn);
 
                 amrex::Gpu::Atomic::Add(&jy_arr(lo.x + i_new + i, lo.y + k_old + k, 0, 0),
-                    wqy * 0.25_rt * sx_new[i] * sz_old[k]);
+                    wqy * 0.25_rt * sxn_szo);
 
                 amrex::Gpu::Atomic::Add(&jy_arr(lo.x + i_old + i, lo.y + k_new + k, 0, 0),
-                    wqy * 0.25_rt * sx_old[i] * sz_new[k]);
+                    wqy * 0.25_rt * sxo_szn);
 
                 amrex::Gpu::Atomic::Add(&jy_arr(lo.x + i_old + i, lo.y + k_old + k, 0, 0),
-                    wqy * 0.25_rt * sx_old[i] * sz_old[k]);
+                    wqy * 0.25_rt * sxo_szo);
 
                 // Jz
                 amrex::Gpu::Atomic::Add(&jz_arr(lo.x + i_new + i, lo.y + k_new + k, 0, 0),
-                    wq * invvol * invdt * 0.5_rt * sx_new[i] * sz_new[k]);
+                    wq * invvol * invdt * 0.5_rt * sxn_szn);
 
                 amrex::Gpu::Atomic::Add(&jz_arr(lo.x+i_new+i,lo.y+k_old+k,0,0),
-                    - wq * invvol * invdt * 0.5_rt * sx_new[i] * sz_old[k]);
+                    - wq * invvol * invdt * 0.5_rt * sxn_szo);
 
                 amrex::Gpu::Atomic::Add(&jz_arr(lo.x+i_old+i,lo.y+k_new+k,0,0),
-                    wq * invvol * invdt * 0.5_rt * sx_old[i] * sz_new[k]);
+                    wq * invvol * invdt * 0.5_rt * sxo_szn);
 
                 amrex::Gpu::Atomic::Add(&jz_arr(lo.x + i_old + i, lo.y + k_old + k, 0, 0),
-                    - wq * invvol * invdt * 0.5_rt * sx_old[i] * sz_old[k]);
+                    - wq * invvol * invdt * 0.5_rt * sxo_szo);
             }
         }
 
 #elif (defined WARPX_DIM_3D)
 
         for (int k=0; k<=depos_order; k++) {
             for (int j=0; j<=depos_order; j++) {
+
+                amrex::Real syn_szn = sy_new[j] * sz_new[k];
+                amrex::Real syo_szn = sy_old[j] * sz_new[k];
+                amrex::Real syn_szo = sy_new[j] * sz_old[k];
+                amrex::Real syo_szo = sy_old[j] * sz_old[k];
+                
                 for (int i=0; i<=depos_order; i++) {
 
+                    amrex::Real sxn_syn_szn = sx_new[i] * syn_szn;
+                    amrex::Real sxo_syn_szn = sx_old[i] * syn_szn;
+                    amrex::Real sxn_syo_szn = sx_new[i] * syo_szn;
+                    amrex::Real sxo_syo_szn = sx_old[i] * syo_szn;
+                    amrex::Real sxn_syn_szo = sx_new[i] * syn_szo;
+                    amrex::Real sxo_syn_szo = sx_old[i] * syn_szo;
+                    amrex::Real sxn_syo_szo = sx_new[i] * syo_szo;
+                    amrex::Real sxo_syo_szo = sx_old[i] * syo_szo;
```

Comment:
```suggestion
                amrex::Real const syn_szn = sy_new[j] * sz_new[k];
                amrex::Real const syo_szn = sy_old[j] * sz_new[k];
                amrex::Real const syn_szo = sy_new[j] * sz_old[k];
                amrex::Real const syo_szo = sy_old[j] * sz_old[k];
                
                for (int i=0; i<=depos_order; i++) {

                    amrex::Real const sxn_syn_szn = sx_new[i] * syn_szn;
                    amrex::Real const sxo_syn_szn = sx_old[i] * syn_szn;
                    amrex::Real const sxn_syo_szn = sx_new[i] * syo_szn;
                    amrex::Real const sxo_syo_szn = sx_old[i] * syo_szn;
                    amrex::Real const sxn_syn_szo = sx_new[i] * syn_szo;
                    amrex::Real const sxo_syn_szo = sx_old[i] * syn_szo;
                    amrex::Real const sxn_syo_szo = sx_new[i] * syo_szo;
                    amrex::Real const sxo_syo_szo = sx_old[i] * syo_szo;
```

Reply:
the comment below is committed which is the same as what is suggested here. (before and after eol fix.) Resolving conversation